### PR TITLE
Lab 10 Trybushnyi Andryii 

### PR DESCRIPTION
**Pull request by roflmyrlok**

### CODE REVIEW

File: lecture_notes/10 refactoring/1-not-optimal.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    return points
+
+polyline = [
+  { "x": 0, "y": 0 },
+  { "x": 10, "y": 10 },
+  '{ "x": 20, "y": 20 }',
+  { "x": 30, "y": 30 },
+]
+
+shift({'x': 10, 'y': -5}, polyline);
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/1-not-optimal.py:21]: Semicolon in python?

File: lecture_notes/10 refactoring/3-optimized.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import json
+
+def parse(point):
+    point = json.loads(point)
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/3-optimized.py:4]: Don't rewrite incoming parameters

File: lecture_notes/10 refactoring/3-optimized.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    return point
+
+def shift(offset, points):
+    for i in range(0, len(points), 1):
+        if type(points[i]) == str:
+            points[i] = parse(points[i])
+        moove(points[i], offset)
+    return points
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/3-optimized.py:12]: Don't rewrite incoming parameters

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
- Try to optimize using map instead of for loop
- Don not modify incoming structures like list and point, create new one instead
- Waiting for js version

File: lecture_notes/10 refactoring/3-optimized.py (hunk 2 above)
Comment (roflmyrlok) [reply lecture_notes/10 refactoring/3-optimized.py:4]: please, check the solution

File: lecture_notes/10 refactoring/1-not-optimal.py (hunk 1 above)
Comment (roflmyrlok) [reply lecture_notes/10 refactoring/1-not-optimal.py:21]: that's why its named "non-optimal"

Comment (roflmyrlok) [reply lecture_notes/10 refactoring/1-not-optimal.py:21]: (fixed)

**Pull request closed without merging**